File: read.js
const fs = require('fs');
const Buffer = require('buffer').Buffer;

const parse = () => {
    fs.readFile('sample.png', (err, data) => {
        if (err) throw err;
        let raw_data = Buffer.from(data)
        let prefix = raw_data.slice(0, 8)
        if (prefix.toString() != Buffer.from('\x89PNG\r\n\x1a\n', 'binary').toString()) {
            throw 'This file is not png'
        }
        console.log(prefix)
        console.log(Buffer.from('\x89PNG\r\n\x1a\n', 'binary'))
    })
}

parse()

File: read.go
package main

import (
	"bytes"
	"fmt"
	"io"
	"os"
)

func parse(r io.Reader) error {
	fmt.Printf("%+v", r)
	buffer := new(bytes.Buffer)
	_, err := buffer.ReadFrom(r)
	if err != nil {
		return err
	}
	fmt.Println(string(buffer.Next(8)))
	if string(buffer.Next(8)) != "\x89PNG\r\n\x1a\n" {
		return fmt.Errorf("not a PNG")
	}

	return nil
}

func main() {
	imageFile := "sub.png"
	file, err := os.Open(imageFile)
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Printf("%+v", file)
	defer file.Close()

	err = parse(file)
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Println("Complete")
}
